plugins/kubernetes-backend/src/service/KubernetesBuilder.ts

```diff
@@ -243,13 +243,28 @@ export class KubernetesBuilder {
       'kubernetes.objectTypes',
     ) as KubernetesObjectTypes[];
 
+    const apiVersionOverrides = this.env.config.getOptionalConfig(
+      'kubernetes.apiVersionOverrides',
+    );
+
     let objectTypesToFetch;
 
     if (objectTypesToFetchStrings) {
       objectTypesToFetch = DEFAULT_OBJECTS.filter(obj =>
         objectTypesToFetchStrings.includes(obj.objectType),
       );
     }
+
+    if (apiVersionOverrides) {
+      objectTypesToFetch = objectTypesToFetch ?? DEFAULT_OBJECTS;
+
+      for (const obj of objectTypesToFetch) {
+        if (apiVersionOverrides.getOptionalString(obj.objectType)) {
```

Comment:
Could use `apiVersionOverrides.has(obj.objectType)` instead here to clarify the intent a bit